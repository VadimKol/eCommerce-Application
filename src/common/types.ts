import type { BaseAddress, CustomFieldsDraft, StoreResourceIdentifier } from '@commercetools/platform-sdk';

export interface AuthContextInterface {
  isAuthenticated: boolean;
  handleLogin: () => void;
  handleLogout: () => void;
}

export type CategoriesContextType = {
  categories: CategoriesData | null;
  loading: boolean;
  error: Error | null;
};

export interface Credentials {
  clientId: string;
  clientSecret: string;
}

export interface User {
  username: string;
  password: string;
}

export interface GeekShopCustomerDraft {
  /**
   *	Email address of the Customer that is [unique](/../api/customers-overview#customer-uniqueness) for an entire Project or Store the Customer is assigned to.
   *	It is the mandatory unique identifier of a Customer.
   *
   *
   */
  readonly email: string;
  /**
   *	Password of the Customer.
   *
   *
   */
  readonly password: string;
  /**
   *	Given name (first name) of the Customer.
   *
   *
   */
  readonly firstName?: string;
  /**
   *	Family name (last name) of the Customer.
   *
   *
   */
  readonly lastName?: string;
  /**
   *	Middle name of the Customer.
   *
   *
   */
  readonly middleName?: string;
  /**
   *	Title of the Customer, for example, 'Dr.'.
   *
   *
   */
  readonly title?: string;
  /**
   *	Salutation of the Customer, for example, 'Mr.' or 'Mrs.'.
   *
   *
   */
  readonly salutation?: string;
  /**
   *	Date of birth of the Customer.
   *
   *
   */
  readonly dateOfBirth?: string;
  /**
   *	Company name of the Customer.
   *
   *
   */
  readonly companyName?: string;
  /**
   *	Individual VAT ID of the Customer.
   *
   *
   */
  readonly vatId?: string;
  /**
   *	Addresses of the Customer.
   *
   *
   */
  readonly addresses?: BaseAddress[];
  /**
   *	Indexes of entries in `addresses` to set as shipping addresses.
   *	The `shippingAddressIds` of the [Customer](ctp:api:type:Customer) will be replaced by these addresses.
   *
   *
   */
  readonly shippingAddresses?: number[];
  /**
   *	Index of the address in the `addresses` array to use as the default shipping address.
   *	The `defaultShippingAddressId` of the Customer will be set to the `id` of that address.
   *
   *
   */
  readonly defaultShippingAddress?: number;
  /**
   *	Indexes of entries in `addresses` to set as billing addresses.
   *	The `billingAddressIds` of the [Customer](ctp:api:type:Customer) will be replaced by these addresses.
   *
   *
   */
  readonly billingAddresses?: number[];
  /**
   *	Index of the address in the `addresses` array to use as the default billing address.
   *	The `defaultBillingAddressId` of the Customer will be set to the `id` of that address.
   *
   *
   */
  readonly defaultBillingAddress?: number;
  /**
   *	Custom Fields for the Customer.
   *
   *
   */
  readonly custom?: CustomFieldsDraft;
  /**
   *	Preferred language of the Customer. Must be one of the languages supported by the [Project](ctp:api:type:Project).
   *
   *
   */
  readonly locale?: string;
  /**
   *	Sets the [Stores](ctp:api:type:Store) for the Customer.
   *
   *
   */
  readonly stores?: StoreResourceIdentifier[];
}

export type ProductDetails = {
  name?: string;
  description?: string;
  price?: string | null;
  discountedPrice?: string | null;
  currency?: string | null;
  images?: string[];
  attributes?: { name: string; value: string | number | string[] }[];
  availability?: {
    isOnStock?: boolean | null;
    availableQuantity?: number | null;
  };
};

export type Product = {
  id: string;
  name: string;
  description: string;
  slug: string;
  key: string;
  sku: string;
  quantity: number;
  price: string;
  images: string[];
  categoryId: string;
  slugCategory: string;
  keyCategory: string;
  subcategoryId: string;
  slugSubCategory: string;
  keySubCategory: string;
};

export type Subcategory = {
  id: string;
  name: string;
  slug: string;
  key: string;
  products: Product[];
};

export type Category = {
  id: string;
  name: string;
  slug: string;
  key: string;
  subcategories: Subcategory[];
};

export type CategoriesData = Category[];

export type UrlParams = {
  categoryName?: string;
  subcategoryName?: string;
  productName?: string;
};

export type LoaderData = {
  category?: Category;
  subcategory?: Subcategory;
  product?: Product;
};
