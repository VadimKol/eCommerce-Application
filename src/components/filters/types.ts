export interface FilterProps {
  priceFilter: [number, number];
  setPriceFilter: (priceFilter: [number, number]) => void;
  franchises: boolean[];
  setFranchises: (franchises: boolean[]) => void;
  setPage: (page: number) => void;
}
