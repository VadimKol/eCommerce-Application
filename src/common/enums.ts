export enum NavigationPaths {
  HOME = '/',
  CATALOG = '/catalog',
  ABOUT = '/about',
}

export enum ActionPaths {
  LOGIN = '/login',
  REGISTER = '/register',
  PROFILE = '/profile',
}
