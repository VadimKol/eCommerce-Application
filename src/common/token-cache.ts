import type { TokenCache, TokenStore } from '@commercetools/sdk-client-v2';

export class CustomTokenCache implements TokenCache {
  private tokenStore: TokenStore = {
    token: localStorage.getItem('geek-shop-token') ? (localStorage.getItem('geek-shop-token') as string) : '',
    expirationTime: 0,
    refreshToken: '',
  };

  public set(tokenStore: TokenStore): void {
    this.tokenStore = tokenStore;
  }

  public get(): TokenStore {
    return this.tokenStore;
  }
}
