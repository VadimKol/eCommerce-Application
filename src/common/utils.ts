import type { CatalogAction, CatalogState } from './types';

export const calculateAge = (birthDate: string): boolean => {
  const today = new Date();
  const birth = new Date(birthDate);
  let age = today.getFullYear() - birth.getFullYear();
  const monthDifference = today.getMonth() - birth.getMonth();
  if (monthDifference < 0 || (monthDifference === 0 && today.getDate() < birth.getDate())) {
    age -= 1;
  }
  return age >= 13;
};

export const sortingTypes = ['Alphabetically, A-Z', 'Alphabetically, Z-A', 'Price, low to high', 'Price, high to low'];

export const fandoms = [
  'Genshin',
  'Disney',
  'Cyberpunk',
  'Batman',
  'Doctor Who',
  'Star Wars',
  'Beetlejuice',
  'Marvel',
  'One Piece',
  'Harry Potter',
  'Stranger Things',
  'Rick and Morty',
  'Supernatural',
];

export const countriesFilter = ['China', 'France', 'Germany', 'India', 'Italy', 'Korea', 'US'];
export const materials = ['Aluminum', 'Cardboard', 'Ceramic', 'Coir', 'Paper', 'Plastic', 'Polyester'];
export const countries = [
  {
    title: 'United States',
    code: 'US',
    errorMsg: 'It must be 5 digits or 5+4 digits',
    postcode: /^\d{5}(-\d{4})?$/,
  },
  {
    title: 'Belarus',
    code: 'BY',
    errorMsg: 'It must be exactly 6 digits.',
    postcode: /^\d{6}$/,
  },
  {
    title: 'Russia',
    code: 'RU',
    errorMsg: 'It must be exactly 6 digits.',
    postcode: /^\d{6}$/,
  },
];

export const DEFAULT_LOCALE = 'en-US';
export const QUERY_LIMIT = 12;
export const PRICE_FILTER_MIN = 1;
export const PRICE_FILTER_MAX = 100;
export const REFRESH_TOKEN_EXPIRATION_DAYS = 150;
export const Discounts = [
  {
    name: 'GEEK-SHOP',
    desc: '-10% cart',
  },
  {
    name: 'COSPLAY',
    desc: '-10% category',
  },
];

export function getCheckboxFilter(checkboxes: boolean[], attribute: string): string {
  const filter: string[] = [];
  let constantsFilter: string[] = [];

  switch (attribute) {
    case 'Fandom': {
      constantsFilter = fandoms;
      break;
    }
    case 'Country': {
      constantsFilter = countriesFilter;
      break;
    }
    case 'Material': {
      constantsFilter = materials;
      break;
    }
    default:
      break;
  }

  checkboxes.forEach((checkbox, index) => {
    constantsFilter.forEach((element, i) => {
      if (index === i && checkbox) {
        filter.push(`"${element}"`);
      }
    });
  });

  return filter.length > 0 ? `variants.attributes.${attribute}:${filter.join(',')}` : '';
}

export function getSort(sortType: string): string | undefined {
  switch (sortType) {
    case 'Alphabetically, A-Z':
      return `name.${DEFAULT_LOCALE} asc`;
    case 'Alphabetically, Z-A':
      return `name.${DEFAULT_LOCALE} desc`;
    case 'Price, low to high':
      return 'price asc';
    case 'Price, high to low':
      return 'price desc';
    default:
      return undefined;
  }
}

export class StatusError extends Error {
  public statusCode: number;

  constructor(message: string, status: number) {
    super(message);
    this.statusCode = status;
  }
}

export const initialState: CatalogState = {
  products: [],
  page: 0,
  sortType: 'Sort',
  search: '',
  priceFilter: [PRICE_FILTER_MIN, PRICE_FILTER_MAX],
  franchises: Array(fandoms.length).fill(false),
  countriesF: Array(countriesFilter.length).fill(false),
  materialsF: Array(materials.length).fill(false),
  loadingProducts: true,
  categories: {},
};

export function reducerCatalog(
  state: CatalogState,
  {
    type,
    products = [],
    page = 0,
    sortType = 'Sort',
    search = '',
    priceFilter = [PRICE_FILTER_MIN, PRICE_FILTER_MAX],
    franchises = Array(fandoms.length).fill(false),
    countriesF = Array(countriesFilter.length).fill(false),
    materialsF = Array(materials.length).fill(false),
    loadingProducts = true,
    categories = {},
  }: CatalogAction,
): CatalogState {
  switch (type) {
    case 'SET_CATEGORIES':
      return {
        ...state,
        categories,
        page,
        sortType,
        priceFilter,
        franchises,
        countriesF,
        materialsF,
        search,
        loadingProducts,
      };
    case 'SET_PRODUCTS':
      return { ...state, products };
    case 'SET_LOADING_PRODUCTS':
      return { ...state, loadingProducts };
    case 'SET_SEARCH':
      return { ...state, search, page, loadingProducts };
    case 'SET_SORT_TYPE':
      return { ...state, sortType, loadingProducts };
    case 'SET_PRICE':
      return { ...state, priceFilter, page, loadingProducts };
    case 'SET_FRANCHISES':
      return { ...state, franchises, page, loadingProducts };
    case 'SET_COUNTRIES':
      return { ...state, countriesF, page, loadingProducts };
    case 'SET_MATERIALS':
      return { ...state, materialsF, page, loadingProducts };
    case 'RESET_FILTERS':
      return { ...state, priceFilter, franchises, countriesF, materialsF, page, loadingProducts };
    case 'SET_PAGE':
      return { ...state, page, loadingProducts };
    default:
      return state;
  }
}

export const capitalizeFirstLetter = (string: string): string => string.charAt(0).toUpperCase() + string.slice(1);
