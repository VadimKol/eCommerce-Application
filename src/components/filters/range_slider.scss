.multi-range-slider-blue * {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.multi-range-slider-blue {
  position: relative;
  display: flex;
  flex-direction: column;
  padding: 20px 10px;
  border: solid 1px #808080;
  border-radius: 10px;
  box-shadow: 1px 1px 4px #000000;
  user-select: none;
  -webkit-touch-callout: none; /* iOS Safari */
}

.multi-range-slider-blue .bar {
  display: flex;
}

.multi-range-slider-blue .bar-left {
  width: 25%;
  padding: 4px 0;
  background-color: #f0f0f0;
  border-radius: 10px 0 0 10px;
  box-shadow: inset 0 0 5px #000000;
}

.multi-range-slider-blue .bar-right {
  width: 25%;
  background-color: #f0f0f0;
  border-radius: 0 10px 10px 0;
  box-shadow: inset 0 0 5px #000000;
}

.multi-range-slider-blue .bar-inner {
  position: relative;
  display: flex;
  flex-grow: 1;
  flex-shrink: 1;
  justify-content: space-between;
  background-color: #00ff00;
  border: solid 1px #000000;
  box-shadow: inset 0 0 5px #000000;
}

.multi-range-slider-blue .bar-inner-left {
  width: 50%;
}

.multi-range-slider-blue .bar-inner-right {
  width: 50%;
}

.multi-range-slider-blue .thumb {
  position: relative;
  z-index: 1;
  background-color: #ffffff;
  cursor: pointer;
}

.multi-range-slider-blue .thumb::before {
  content: "";
  position: absolute;
  z-index: 1;
  width: 20px;
  height: 20px;
  margin: -8px -12px;
  background-color: inherit;
  border: solid 1px var(--primary-color);
  border-radius: 50%;

  // box-shadow: 0 0 3px #000000, inset 0 0 5px #808080;
  cursor: pointer;
}

.multi-range-slider-blue .input-type-range:focus + .thumb::after {
  content: "";
  position: absolute;
  top: -4px;
  left: -7px;
  z-index: 2;
  width: 11px;
  height: 11px;
  border: dotted 1px #000000;
  border-radius: 50%;
  box-shadow: 0 0 5px #ffffff, inset 0 0 10px #000000;
}

.multi-range-slider-blue .caption {
  position: absolute;
  bottom: 35px;
  left: 1px;
  display: none;
  justify-content: center;
  align-items: center;
  overflow: visible;
  width: 2px;
  height: 2px;
}

.multi-range-slider-blue .thumb .caption * {
  position: absolute;
  min-width: 30px;
  height: 30px;
  padding: 0 5px;
  font-size: 75%;
  line-height: 30px;
  text-align: center;
  white-space: nowrap;
  color: #ffffff;
  background-color: #0000ff;
  border-radius: 15px;
  box-shadow: 0 0 5px #000000;
}

/* .multi-range-slider-blue .thumb:active .caption {
  display: flex;
}

.multi-range-slider-blue .input-type-range:focus + .thumb .caption {
  display: flex;
} */

.multi-range-slider-blue .input-type-range {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  opacity: 0;
  pointer-events: none;
}

.multi-range-slider-blue .ruler {
  display: flex;
  overflow: hidden;
  margin: 10px 0 -5px;
}

.multi-range-slider-blue .ruler .ruler-rule {
  display: flex;
  flex-grow: 1;
  flex-shrink: 1;
  padding: 5px 0;
  border-bottom: solid 1px;
  border-left: solid 1px;
}

.multi-range-slider-blue .ruler .ruler-rule:last-child {
  border-right: solid 1px;
}

.multi-range-slider-blue .ruler .ruler-sub-rule {
  bottom: 0;
  display: flex;
  flex-grow: 1;
  flex-shrink: 1;
  margin-bottom: -5px;
  padding: 3px 0;
  border-left: solid 1px;
}

.multi-range-slider-blue .ruler .ruler-sub-rule:first-child {
  border-left: none;
}

.multi-range-slider-blue .labels {
  display: flex;
  justify-content: space-between;
  margin-top: 10px;
  margin-bottom: -20px;
  padding: 0;
}

.multi-range-slider-blue .label {
  display: flex;
  justify-content: center;
  width: 1px;
  font-size: 80%;
}

.multi-range-slider-blue .label:first-child {
  justify-content: start;
}

.multi-range-slider-blue .label:last-child {
  justify-content: end;
}

.multi-range-slider-blue.zero-ranage-margin .thumb-left {
  right: 12px;
}

.multi-range-slider-blue.zero-ranage-margin .thumb-right {
  left: 8px;
}


.multi-range-slider-blue.disabled {
  color: rgb(180 180 180);
  border: solid 1px rgb(200 200 200);
  box-shadow: 1px 1px 4px rgb(180 180 180);
}

.multi-range-slider-blue.disabled .bar-left {
  background-color: #c9c9c9;
  box-shadow: inset 0 0 5px rgb(160 160 160);
}

.multi-range-slider-blue.disabled .bar-right {
  background-color: #c9c9c9;
  box-shadow: inset 0 0 5px rgb(160 160 160);
}

.multi-range-slider-blue.disabled .bar-inner {
  background-color: rgb(130 243 130);
  border: solid 1px rgb(149 149 149);
  box-shadow: inset 0 0 5px rgb(103 103 103);
}

.multi-range-slider-blue.disabled .thumb {
  background-color: #ffffff;
}

.multi-range-slider-blue.disabled .thumb::before {
  border: solid 1px rgb(200 200 200);
  box-shadow: 0 0 3px rgb(35 35 35), inset 0 0 5px #808080;
}

.multi-range-slider-blue.disabled .input-type-range:focus + .thumb::after {
  border: dotted 1px rgb(35 35 35);
  box-shadow: 0 0 5px #ffffff, inset 0 0 10px rgb(35 35 35);
}

.multi-range-slider-blue.disabled .thumb .caption * {
  color: rgb(199 199 199);
  background-color: rgb(84 84 137);
  box-shadow: 0 0 5px rgb(35 35 35);
}

