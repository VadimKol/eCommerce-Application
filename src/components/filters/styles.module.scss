.accordeon,
.checkboxesList {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.accordeon {
  margin-bottom: 20px;
}

.range,
.checkboxesList {
  overflow: hidden;
  max-height: 0;
  padding: 0 15px;
  transition: max-height 0.5s;
}

.range {
  &_active {
    max-height: 78px;
  }
}

.prices {
  display: flex;
  justify-content: space-between;
  font-size: 0.83em;
  font-weight: 100;
}

.checkboxesList {
  font-weight: 100;
  text-align: left;

  &_active {
    max-height: 576px;
  }
}

.item {
  display: flex;
  gap: 10px;
  align-items: center;
  padding: 0 15px;

  @media (hover: hover) {
    &:hover {
      cursor: pointer;
    }
  }
}

.title {
  display: inline-block;
}

.icon {
  display: inline-block;
  width: 20px;
  height: 20px;
  background-image: url("/src/assets/images/arrow_down.svg");
  background-size: cover;
  transition: transform 0.5s;

  &_show {
    transform: rotate(180deg);
  }
}

.checkbox {
  width: 16px;
  height: 16px;
}

.checkbox_label,
.checkbox {
  @media (hover: hover) {
    &:hover {
      cursor: pointer;
    }
  }
}

.apply {
  padding: 0.3em 1.2em;
}
