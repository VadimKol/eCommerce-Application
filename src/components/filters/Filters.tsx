import { useState } from 'react';

// import { apiRoot } from '@/api/build-client';
import { fandoms } from '@/common/utils';

import { CustomButton } from '../custom-button/Ð¡ustomButton';
import { RangeSlider } from '../range-slider/RangeSlider';
import styles from './styles.module.scss';

export function Filters(): JSX.Element {
  const [isPrice, setIsPrice] = useState(false);
  const [isFranchise, setIsFranchise] = useState(false);

  return (
    <>
      <ul className={styles.accordeon}>
        <li>
          <div
            className={styles.item}
            onClick={() => {
              setIsPrice(!isPrice);
            }}
            role="listbox"
            tabIndex={0}
            onKeyUp={() => {}}
          >
            <span className={styles.title}>Price</span>
            <span className={isPrice ? `${styles.icon} ${styles.icon_show}` : styles.icon} />
          </div>
          <div className={isPrice ? `${styles.range} ${styles.range_active}` : styles.range}>
            <RangeSlider min={1} max={150} />
          </div>
        </li>
        <li>
          <div
            className={styles.item}
            onClick={() => {
              setIsFranchise(!isFranchise);
            }}
            role="listbox"
            tabIndex={0}
            onKeyUp={() => {}}
          >
            <span className={styles.title}>Franchise</span>
            <span className={isFranchise ? `${styles.icon} ${styles.icon_show}` : styles.icon} />
          </div>
          <ul
            className={isFranchise ? `${styles.checkboxesList} ${styles.checkboxesList_active}` : styles.checkboxesList}
          >
            {fandoms.map((fandom) => (
              <li key={fandom}>
                <label className={styles.checkbox_label} htmlFor={fandom}>
                  <input className={styles.checkbox} type="checkbox" name="" id={fandom} /> {fandom}
                </label>
              </li>
            ))}
          </ul>
        </li>
      </ul>
      <CustomButton className={styles.apply}>Apply</CustomButton>
    </>
  );
}

/* apiRoot
  .productProjections()
  .search()
  .get({ queryArgs: { sort: 'name.en-US ASC' } })
  .execute()
  .then(console.log)
  .catch(console.error); */
