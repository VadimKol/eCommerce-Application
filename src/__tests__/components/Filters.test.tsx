import { render } from '@testing-library/react';

import { countriesFilter, fandoms, materials, PRICE_FILTER_MAX, PRICE_FILTER_MIN } from '@/common/utils';
import { Filters } from '@/components/filters/Filters';

it('renders correctly', () => {
  const someDispatch = (): void => {};
  const { container } = render(
    <Filters
      priceFilter={[PRICE_FILTER_MIN, PRICE_FILTER_MAX]}
      franchises={Array(fandoms.length).fill(false)}
      countriesF={Array(countriesFilter.length).fill(false)}
      materialsF={Array(materials.length).fill(false)}
      dispatch={someDispatch}
    />,
  );

  expect(container).toMatchSnapshot();
});
