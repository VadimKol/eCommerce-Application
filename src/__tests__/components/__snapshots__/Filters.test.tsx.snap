// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <ul
    class="accordeon"
  >
    <li>
      <div
        class="item"
        role="listbox"
        tabindex="0"
      >
        <span
          class="title"
        >
          Price
        </span>
        <span
          class="icon"
        />
      </div>
      <div
        class="range"
      >
        <div
          class="container"
        >
          <input
            class="thumb thumb--left"
            max="100"
            min="1"
            style="z-index: 5;"
            type="range"
            value="1"
          />
          <input
            class="thumb thumb--right"
            max="100"
            min="1"
            type="range"
            value="100"
          />
          <div
            class="slider"
          >
            <div
              class="slider__track"
            />
            <div
              class="slider__range"
              style="left: 0%; width: 100%;"
            />
            <div
              class="slider__left-value"
            >
              $
              1
            </div>
            <div
              class="slider__right-value"
            >
              $
              100
            </div>
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        class="item"
        role="listbox"
        tabindex="0"
      >
        <span
          class="title"
        >
          Franchise
        </span>
        <span
          class="icon"
        />
      </div>
      <ul
        class="checkboxesList"
      >
        <li>
          <label
            class="checkbox_label"
            for="Genshin"
          >
            <input
              class="checkbox"
              id="Genshin"
              type="checkbox"
            />
             Genshin
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Disney"
          >
            <input
              class="checkbox"
              id="Disney"
              type="checkbox"
            />
             Disney
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Cyberpunk"
          >
            <input
              class="checkbox"
              id="Cyberpunk"
              type="checkbox"
            />
             Cyberpunk
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Batman"
          >
            <input
              class="checkbox"
              id="Batman"
              type="checkbox"
            />
             Batman
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Doctor Who"
          >
            <input
              class="checkbox"
              id="Doctor Who"
              type="checkbox"
            />
             Doctor Who
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Star Wars"
          >
            <input
              class="checkbox"
              id="Star Wars"
              type="checkbox"
            />
             Star Wars
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Beetlejuice"
          >
            <input
              class="checkbox"
              id="Beetlejuice"
              type="checkbox"
            />
             Beetlejuice
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Marvel"
          >
            <input
              class="checkbox"
              id="Marvel"
              type="checkbox"
            />
             Marvel
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="One Piece"
          >
            <input
              class="checkbox"
              id="One Piece"
              type="checkbox"
            />
             One Piece
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Harry Potter"
          >
            <input
              class="checkbox"
              id="Harry Potter"
              type="checkbox"
            />
             Harry Potter
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Stranger Things"
          >
            <input
              class="checkbox"
              id="Stranger Things"
              type="checkbox"
            />
             Stranger Things
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Rick and Morty"
          >
            <input
              class="checkbox"
              id="Rick and Morty"
              type="checkbox"
            />
             Rick and Morty
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Supernatural"
          >
            <input
              class="checkbox"
              id="Supernatural"
              type="checkbox"
            />
             Supernatural
          </label>
        </li>
      </ul>
    </li>
    <li>
      <div
        class="item"
        role="listbox"
        tabindex="0"
      >
        <span
          class="title"
        >
          Country
        </span>
        <span
          class="icon"
        />
      </div>
      <ul
        class="checkboxesList"
      >
        <li>
          <label
            class="checkbox_label"
            for="China"
          >
            <input
              class="checkbox"
              id="China"
              type="checkbox"
            />
             China
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="France"
          >
            <input
              class="checkbox"
              id="France"
              type="checkbox"
            />
             France
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Germany"
          >
            <input
              class="checkbox"
              id="Germany"
              type="checkbox"
            />
             Germany
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="India"
          >
            <input
              class="checkbox"
              id="India"
              type="checkbox"
            />
             India
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Italy"
          >
            <input
              class="checkbox"
              id="Italy"
              type="checkbox"
            />
             Italy
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Korea"
          >
            <input
              class="checkbox"
              id="Korea"
              type="checkbox"
            />
             Korea
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="US"
          >
            <input
              class="checkbox"
              id="US"
              type="checkbox"
            />
             US
          </label>
        </li>
      </ul>
    </li>
    <li>
      <div
        class="item"
        role="listbox"
        tabindex="0"
      >
        <span
          class="title"
        >
          Material
        </span>
        <span
          class="icon"
        />
      </div>
      <ul
        class="checkboxesList"
      >
        <li>
          <label
            class="checkbox_label"
            for="Aluminum"
          >
            <input
              class="checkbox"
              id="Aluminum"
              type="checkbox"
            />
             Aluminum
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Cardboard"
          >
            <input
              class="checkbox"
              id="Cardboard"
              type="checkbox"
            />
             Cardboard
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Ceramic"
          >
            <input
              class="checkbox"
              id="Ceramic"
              type="checkbox"
            />
             Ceramic
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Coir"
          >
            <input
              class="checkbox"
              id="Coir"
              type="checkbox"
            />
             Coir
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Paper"
          >
            <input
              class="checkbox"
              id="Paper"
              type="checkbox"
            />
             Paper
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Plastic"
          >
            <input
              class="checkbox"
              id="Plastic"
              type="checkbox"
            />
             Plastic
          </label>
        </li>
        <li>
          <label
            class="checkbox_label"
            for="Polyester"
          >
            <input
              class="checkbox"
              id="Polyester"
              type="checkbox"
            />
             Polyester
          </label>
        </li>
      </ul>
    </li>
  </ul>
  <button
    class="button button--primary reset"
    id=""
    type="button"
  >
    Reset
  </button>
</div>
`;
